src/io/read.rs

```diff
@@ -59,6 +59,9 @@ where
             }
         }
 
-        (res, buf)
+        match res {
```

Comment:
Actually, we've maintained an adaptor trait for this 

```rust
/// Adapter trait to convert Result<T, E> to BufResult<T, B> where E can be
/// converted to std::io::Error.
pub trait WithBuf<T, B>: Sized {
    fn with_buf(self, buf: B) -> T;
}

/// Adaptor implementation for Result<T, E> to BufResult<T, B>.
impl<T, B, E> WithBuf<BufResult<T, B>, B> for Result<T, E>
where
    E: Into<std::io::Error>,
{
    fn with_buf(self, buf: B) -> BufResult<T, B> {
        match self {
            Ok(res) => Ok((res, buf)),
            Err(err) => Err(BufError(err.into(), buf)),
        }
    }
}
```

then, use it here

```rust
res.with_buf(buf)
```

I think it's very common to convert `std::io::Result` to `tokio_uring::Result`, so glad to see the implementation in tokio-uring itself.

Reply:
That would certainly remove a heap of boilerplate. I'll add that as a sealed trait and tidy those instances up.